tools/c7n_gcp/c7n_gcp/filters/labels.py

```diff
@@ -78,6 +80,17 @@ def process(self, resources, event=None):
         self.tz = Time.get_tz(self.data.get('tz', 'utc'))
         return super(LabelActionFilter, self).process(resources, event)
 
+    def parse(self, v: str):
+        remainder, action_date = v.rsplit('-', 1)
+        found = False
+        msg = ""
+        for a in self.valid_actions:
+            if remainder.endswith(a):
```

Comment:
That's handy. Looks like this check might get awkward if we had multiple valid actions with the same ending (like `start` and `restart` or something). But since afaics we don't... that's a nit.

Reply:
we sort the list by length, so its longest prefix search in decreasing size of prefix to handle the ambiguity on similar start/ends of words.

Reply:
Oh nice I totally missed that, awesome 👍